src/events.js

```diff
@@ -96,12 +94,14 @@ module.exports = function(ctx) {
     else if (event.keyCode === 51) {
       ctx.api.changeMode('draw_polygon');
     }
+    ctx.ui.fireClassUpdate();
   };
 
   events.keyup = function(event) {
     if (isKeyModeValid(event.keyCode)) {
       currentMode.keyup(event);
     }
+    ctx.ui.fireClassUpdate();
```

Comment:
You should be able to drop this as this will be caught by the mode_handler delegator.